scylla/src/transport/authenticate_test.rs

```diff
@@ -21,3 +25,60 @@ async fn authenticate_superuser() {
 
     println!("Ok.");
 }
+
+struct CustomAuthenticator;
+
+#[async_trait]
+impl AuthenticatorSession for CustomAuthenticator {
+    async fn evaluate_challenge(
+        &mut self,
+        _token: Option<&[u8]>,
+    ) -> Result<Option<Vec<u8>>, AuthError> {
+        Err("Challenges are not expected".to_string())
+    }
+
+    async fn success(&mut self, _token: Option<&[u8]>) -> Result<(), AuthError> {
+        Ok(())
+    }
+}
+
+struct CustomAuthenticatorProvider;
+
+#[async_trait]
+impl AuthenticatorProvider for CustomAuthenticatorProvider {
+    async fn start_authentication_session(
+        &self,
+        _authenticator_name: &str,
+    ) -> Result<(Option<Vec<u8>>, Box<dyn AuthenticatorSession>), AuthError> {
+        let mut response = BytesMut::new();
+        let cred = "\0cassandra\0cassandra";
+        let cred_length = 20;
+
+        response.put_i32(cred_length);
```

Comment:
I assume that the custom authenticator here works the same as the plain text authenticator. Like in the case of the `PlainTextAuthenticator`, you shouldn't serialize the length here - it's not a part of the response token, and `AuthResponse::serialize` takes care of serializing the length of the token.